csireverseproxy/pkg/config/config.go

```diff
@@ -18,23 +18,49 @@ import (
 	"crypto/subtle"
 	"fmt"
 	"net/url"
+	"os"
+	"path/filepath"
 	"reflect"
+	"strconv"
 
-	"revproxy/v2/pkg/common"
-	"revproxy/v2/pkg/k8sutils"
-	"revproxy/v2/pkg/utils"
+	"github.com/dell/csi-powermax/csireverseproxy/v2/pkg/common"
+	"github.com/dell/csi-powermax/csireverseproxy/v2/pkg/k8sutils"
+	"github.com/dell/csi-powermax/csireverseproxy/v2/pkg/utils"
 
+	"github.com/mitchellh/mapstructure"
 	log "github.com/sirupsen/logrus"
 
 	"github.com/spf13/viper"
 	corev1 "k8s.io/api/core/v1"
 )
 
+// ConfigManager is an interface used for testing, satisfied by viper.Viper.
+//
+//go:generate mockgen -source=config.go -destination=mocks/config-manager.go
+type ConfigManager interface {
+	// SetConfigFile designates the name of the file containing the configuration
+	SetConfigName(string)
+
+	// SetConfigType designates the type of the configuration. e.g. yaml, json
+	SetConfigType(string)
+
+	// AddConfigPath adds a path to look for the config file in
+	// Can be called multiple times to define multiple search paths
+	AddConfigPath(string)
+
+	// ReadInConfig will discover and load the configuration file from disk
+	// and key/value stores, searching in one of the defined paths.
+	ReadInConfig() error
+
+	// GetString returns the value associated with the key as a string.
+	GetString(string) string
+}
+
 // StorageArrayConfig represents the configuration of a storage array in the config file
 type StorageArrayConfig struct {
 	StorageArrayID         string   `yaml:"storageArrayId"`
-	PrimaryURL             string   `yaml:"primaryURL"`
-	BackupURL              string   `yaml:"backupURL,omitempty"`
+	PrimaryEndpoint        string   `yaml:"primaryEndpoint"`
```

Comment:
Does this rename break upgrade when continuing to use the existing ConfigMap and not switching over to a Secret?

Reply:
It should not break existing configMap. There is custom unmarshal now that should handle this.